File: unused/Courses.py

class API():
    def __init__(self, TRANSLATOR, TTS, STT):
        self.TRANSLATOR = TRANSLATOR    # Preferred Translator API
        self.TTS = TTS                  # Preferred Text-to-Speech API
        self.STT = STT                  # Preferred Speech-to-Text API

class lang():
    def __init__(self, NATIVE, TARGET):
        self.NATIVE = NATIVE    # Two-letter Abbreviation for the Native Language
        self.TARGET = TARGET    # Two-letter Abbreviation for the Target Language

class settings():
    def __init__(self, API, LANG):
        self.API = API      # API Class containing API Selections
        self.LANG = LANG    # Language Class containing Native and Target Languages

class image():
    def __init__(self, PATH, NATIVETEXT, TARGETTEXT):
        self.PATH = PATH                # Path to image file
        self.NATIVETEXT = NATIVETEXT    # Description of image in Native Language
        self.TARGETTEXT = TARGETTEXT    # Description of image in Target Language

class media():
    def __init__(self, NATIVESENTENCES, NATIVEAUDIO, TARGETSENTENCES, TARGETAUDIO, IMAGES):
        self.NATIVESENTENCES = NATIVESENTENCES  # List of Native Sentences
        self.NATIVEAUDIO = NATIVEAUDIO          # List of Audio File Paths
        self.TARGETSENTENCES = TARGETSENTENCES  # List of Target Sentences
        self.TARGETAUDIO = TARGETAUDIO          # List of Audio File Paths
        self.IMAGES = IMAGES                    # List of Image Classes

class lesson():
    def __init__(self, TITLE, ICON, CHECKPOINT, LEVEL, MEDIA):
        self.TITLE = TITLE              # Lesson Title
        self.ICON = ICON                # Icon Name (Default is Material Design Icons)
        self.CHECKPOINT = CHECKPOINT    # Checkpoint of Lesson (STARTS AT 0)
        self.LEVEL = LEVEL              # Level Within Checkpoint (STARTS AT 0)
        self.MEDIA = MEDIA              # Media Class containing Sentences, Audio, and Images

class course():
    def __init__(self, TITLE, SETTINGS, LESSONS):
        self.TITLE = TITLE          # Course Title
        self.SETTINGS = SETTINGS    # Settings
        self.LESSONS = LESSONS      # Lessons

File: src/Javascript/Courses.mjs

// Outdated JS File for Courses- Kept for Documentation

class API {
    constructor(TRANSLATOR, TTS, STT) {
        this.TRANSLATOR = TRANSLATOR    // Preferred Translator API
        this.TTS = TTS                  // Preferred Text-to-Speech API
        this.STT = STT                  // Preferred Speech-to-Text API
    }
}

class lang {
    constructor(NATIVE, TARGET, NATIVENAME, TARGETNAME) {
        this.NATIVE = NATIVE    // Two-letter Abbreviation for the Native Language
        this.TARGET = TARGET    // Two-letter Abbreviation for the Target Language
        this.NATIVENAME = NATIVENAME    // Two-letter Abbreviation for the Native Language
        this.TARGETNAME = TARGETNAME    // Two-letter Abbreviation for the Target Language
    }
}

class settings {
    constructor(API, LANG) {
        this.API = API      // API Class containing API Selections
        this.LANG = LANG    // Language Class containing Native and Target Languages
    }
}

class image {
    constructor(PATH, NATIVETEXT, TARGETTEXT) {
        this.PATH = PATH                // Path to image file
        this.NATIVETEXT = NATIVETEXT    // Description of image in Native Language
        this.TARGETTEXT = TARGETTEXT    // Description of image in Target Language
    }
}

class mediasettings {
    constructor(TEXTTASK, AUDIOTASK, AUDIOTYPE) {
        this.TEXTTASK = TEXTTASK        // Text Task Status
        this.AUDIOTASK = AUDIOTASK      // Audio Task Status
        this.AUDIOTYPE = AUDIOTYPE      // String of Audio Type (custom/tts/none)
    }
}

class media {
    constructor(MEDIASETTINGS, NATIVESENTENCES, NATIVEAUDIO, TARGETSENTENCES, TARGETAUDIO, IMAGES) {
        this.MEDIASETTINGS = MEDIASETTINGS
        this.NATIVESENTENCES = NATIVESENTENCES  // List of Native Sentences
        this.NATIVEAUDIO = NATIVEAUDIO          // List of Audio File Paths
        this.TARGETSENTENCES = TARGETSENTENCES  // List of Target Sentences
        this.TARGETAUDIO = TARGETAUDIO          // List of Audio File Paths
        this.IMAGES = IMAGES                    // List of Image Classes
    }
}

class lesson {
    constructor(TITLE, ICON, CHECKPOINT, LEVEL, MEDIA) {
        this.TITLE = TITLE              // Lesson Title
        this.ICON = ICON                // Icon Name (Default is Material Design Icons)
        this.CHECKPOINT = CHECKPOINT    // Checkpoint of Lesson (STARTS AT 0)
        this.LEVEL = LEVEL              // Level Within Checkpoint (STARTS AT 0)
        this.MEDIA = MEDIA              // Media Class containing Sentences, Audio, and Images
    }
}

class course {
    constructor(TITLE, SETTINGS, LESSONS) {
        this.TITLE = TITLE          // Course Title
        this.SETTINGS = SETTINGS    // Settings
        this.LESSONS = LESSONS      // Lessons
    }
}

export { API, lang, settings, image, mediasettings, media, lesson, course }
